libobs/util/platform-cocoa.m

```diff
@@ -417,3 +420,89 @@ uint64_t os_get_proc_virtual_size(void)
 		return 0;
 	return taskinfo.virtual_size;
 }
+
+/* Obtains a copy of the contents of a CFString in specified encoding.
+ * Returns char* (must be bfree'd by caller) or NULL on failure.
+ */
+char* cfstr_copy_cstr(CFStringRef cfstring, CFStringEncoding cfstring_encoding)
+{
+	if (!cfstring) return NULL;
+
+	// Try the quick way to obtain the buffer
+	const char *tmp_buffer = CFStringGetCStringPtr(cfstring, cfstring_encoding);
+
+	if (tmp_buffer != NULL) {
+		return bstrdup(tmp_buffer);
+	}
+
+	// The quick way did not work, try the more expensive one
+	CFIndex length = CFStringGetLength(cfstring);
+	CFIndex max_size =
+		CFStringGetMaximumSizeForEncoding(length, cfstring_encoding);
+
+	// If result would exceed LONG_MAX, kCFNotFound is returned
+	if (max_size == kCFNotFound) {
+		return NULL;
+	}
+
+	// Account for the null terminator
+	max_size++;
+
+	char *buffer = bmalloc(max_size);
+
+	if (buffer == NULL) {
+		return NULL;
+	}
+
+	// Copy CFString in requested encoding to buffer
+	Boolean success =
```

Comment:
I think "boolean" is usually lowercased.  Unless this is special somehow?

Reply:
`Boolean` is a special macOS thing, afaik. It is defined in `MacTypes.h` and written with an uppercase `B`.